File: content-solutions/heap/top-words.java
import java.util.*;

class Solution {
    public List<String> solution(String[] words, int k) {
        Map<String, Integer> count = new HashMap<>();
        for (String w : words) count.put(w, count.getOrDefault(w, 0) + 1);
        PriorityQueue<String> h = new PriorityQueue<>((a, b) -> {
            int ca = count.get(a), cb = count.get(b);
            return ca != cb ? cb - ca : a.compareTo(b);
        });
        h.addAll(count.keySet());
        List<String> answer = new ArrayList<>();
        for (int i = 0; i < k; i++) answer.add(h.poll());
        return answer;
    }
}

File: content-solutions/heap/top-words.py
import heapq


def solution(words, k):
    count = {}
    for w in words:
        count[w] = count.get(w, 0) + 1
    h = [(-c, w) for w, c in count.items()]
    heapq.heapify(h)
    return [heapq.heappop(h)[1] for _ in range(k)]
